function-invokers/java-function-invoker/src/main/java/io/sk8s/invoker/java/server/FunctionConfiguration.java

```diff
@@ -0,0 +1,141 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.sk8s.invoker.java.server;
+
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnMissingBean;
+import org.springframework.boot.context.properties.ConfigurationProperties;
+import org.springframework.cloud.deployer.resource.maven.MavenProperties;
+import org.springframework.cloud.deployer.resource.maven.MavenResource;
+import org.springframework.cloud.deployer.resource.maven.MavenResourceLoader;
+import org.springframework.cloud.deployer.resource.support.DelegatingResourceLoader;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.core.annotation.Order;
+import org.springframework.core.env.Environment;
+import org.springframework.core.io.ResourceLoader;
+import org.springframework.stereotype.Component;
+import org.springframework.util.ClassUtils;
+
+/**
+ * Sets up infrastructure capable of instantiating a "functional" bean (whether Supplier,
+ * Function or Consumer) loaded dynamically according to {@link FunctionProperties}.
+ *
+ * <p>
+ * In case of multiple class names, composes functions together.
+ * </p>
+ *
+ * @author Eric Bottard
+ * @author Mark Fisher
+ * @author Dave Syer
+ */
+@Configuration
+public class FunctionConfiguration {
+
+	@Bean
+	@ConfigurationProperties("maven")
+	public MavenProperties mavenProperties() {
+		return new MavenProperties();
+	}
+
+	@Bean
+	@ConditionalOnMissingBean(DelegatingResourceLoader.class)
+	public DelegatingResourceLoader delegatingResourceLoader(
+			MavenProperties mavenProperties) {
+		Map<String, ResourceLoader> loaders = new HashMap<>();
+		loaders.put(MavenResource.URI_SCHEME, new MavenResourceLoader(mavenProperties));
+		return new DelegatingResourceLoader(loaders);
+	}
+
+	/**
+	 * Registers a singleton bean for each of the function classes passed into the
+	 * {@link FunctionProperties}. They are named sequentially "function0", "function1",
+	 * etc. The instances are created in an isolated class loader, so the jar they are
+	 * packed in has to define all the dependencies (except core JDK).
+	 * 
+	 * TODO: Add reactor and maybe Message (not spring-messaging) to class loader in a way
```

Comment:
I just hit this same requirement as I'm working on a `Flux`-based Function that I intend to use in a sample. Do you already have some thoughts on sharing that dependency?

Reply:
I know how to do it with a more complicated class loader structure (c.f. spring-cloud-function-deployer). I think we don't have to solve that problem right here, right now, though?

Reply:
I am building a demo for S1P where the "user" Function has `Flux` for input and output, so we do need to solve at least that part. I am able to make it work from this branch by passing the parent classloader (where you explicitly pass `null`). I know that crude approach is not desirable, but maybe for now we could just have a parent classloader that only shares `reactor-core` and `reactivestreams`. WDYT?

Reply:
Just to be clear, it could be a follow-up PR to this one, but I just want to have a plan so we can have something to support that use-case ASAP.

Reply:
I think that approach might sometimes be useful. You could also go with a simple single classloader approach. If you want we can talk about options and tradeoffs out of band.